tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1553,6 +1569,44 @@ bool StorageManager::is_vacuum_file(const URI& uri) const {
   return false;
 }
 
+Status StorageManager::get_array_schema_uris(
+    const URI& array_uri, std::vector<URI>* uris) const {
+  auto timer_se = stats_->start_timer("read_get_array_schema_uris");
+
+  // Always add array_name/__array_schema.tdb as the first schema uri
+  uris->clear();
+  auto schema_uri_old = array_uri.join_path(constants::array_schema_filename);
+  uris->emplace_back(schema_uri_old);
+
+  // Add schema uris under directory array_name/__schema
+  const URI schema_uri =
+      array_uri.join_path(constants::array_schema_folder_name);
+
+  // Since ls could return NOT Ok status, we will not use RETURN_NOT_OK here
+  vfs_->ls(schema_uri, uris);
+
+  return Status::Ok();
+}
+
+Status StorageManager::get_latest_array_schema_uri(
+    const URI& array_uri, URI* uri) const {
+  auto timer_se = stats_->start_timer("read_get_latest_array_schema_uri");
+
+  std::vector<URI> uris;
+  RETURN_NOT_OK(get_array_schema_uris(array_uri, &uris));
+  if (uris.size() == 0) {
+    return LOG_STATUS(Status::StorageManagerError(
+        "Can not get the latest array schema; Empty array schemas."));
```

Comment:
Please change comment to `Cannot get the latest array schema; No array schemas found.`

Reply:
Ok